app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,46 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(benchmark_run_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+
+    category = benchmark_run.result.keys[0]
+
+    previous_benchmark_runs = BenchmarkRun.where('created_at < ?', benchmark_run.created_at).where(initiator_type: Commit, benchmark_type: benchmark_run.benchmark_type,
```

Comment:
![screenshot from 2016-07-22 11-33-35](https://cloud.githubusercontent.com/assets/11133421/17048000/33441054-5000-11e6-8338-452c7e37da10.png)

Say, for this database of benchmark_runs, we need all the results for benchmark category `ao_bench` then we need to give initiator type, benchmark_type and benchmark_result_type to fetch our required result of `"0.22"`. This way we get all the 1000 results. Giving these queries we are already in **the right repository**.